src/Apps/Artist/routes.tsx

```diff
@@ -24,36 +70,51 @@ export const routes: RouteConfig[] = [
     path: "/artist/:artistID",
     Component: ArtistApp,
     query: graphql`
-      query routes_ArtistTopLevelQuery($artistID: String!) {
+      query routes_ArtistTopLevelQuery($artistID: String!) @raw_response_type {
         artist(id: $artistID) @principalField {
           ...ArtistApp_artist
+          ...routes_Artist @relay(mask: false)
         }
       }
     `,
+    render: ({ Component, props, match }) => {
+      if (Component && props) {
+        const { artist } = props as any
+
+        if (!artist) {
+          return null
+        }
+
+        const showArtistInsights =
+          showMarketInsights(artist) ||
+          (artist.insights && artist.insights.length > 0)
+        const hasArtistContent = hasOverviewContent(artist)
+
+        const alreadyAtWorksForSalePath = match.location.pathname.includes(
```

Comment:
This is where we determine our zero-state routing logic.

If the artist has no "overview-worthy" content, we redirect to the `/works-for-sale` path _unless_ you're already at that path (to avoid infinite looping).